### DISCUSSION TITLE
FILAMENT_MOTION_SENSOR

### DISCUSSION THREAD

**Discussion by Woodcat64** [Jyers/Marlin#779] [Ideas]
Would it be possible to have on/off switch in UI for the optical filament runout encoder style runout sensor? I know I can compile it myself, but perhaps I'm not the only one who is thinking of getting one. Or am I just wasting time since they don't work well?

**Comment by smooreace** [comment]
Filament sensors work perfectly fine. Not sure why I would ever want to
disable it though. Kinda defeats the purpose, unless there is a use that Im
unaware of I suppose.

**Reply by Woodcat64** [reply]
I'm talking about filament **motion** sensors here, which unlike the regular runout sensors can also detect jammed spool and in some cases blocked nozzle or skipping extruder. Also, having on/off switch in UI would mean no compiling every time there is a update and just using pre-compiled binaries.

**Comment by Jyers** [comment]
Unfortunately, the motion sensor option is compile time only

**Reply by Woodcat64** [reply]
Oh well. Thanks anyway, this FW is fantastic!
